enclave_manager/avalon_enclave_manager/enclave_manager.py

```diff
@@ -475,17 +508,35 @@ def start_enclave_manager(config):
 
     try:
         sleep_interval = int(config["EnclaveManager"]["sleep_interval"])
+        key_refresh_interval = \
+            int(config["WorkerKeyRefresh"]["key_refresh_interval"])
     except Exception as err:
         logger.error("Failed to get sleep interval from config file. " +
                      "Setting sleep interval to 10 seconds: %s", str(err))
         sleep_interval = 10
 
+    # key_refresh_time is the time elapsed since last key refresh
+    key_refresh_time = 0
     try:
         while True:
+            if key_refresh_interval > 0 and \
+                    key_refresh_time >= key_refresh_interval:
+                logger.info("Initiate Worker Key refresh based on timer")
+                updated_enclave_signup_data = \
+                    initiate_key_refresh(enclave_manager.enclave_data)
+                # Update Enclave Manager with updated signup data
+                # after key refresh
+                enclave_manager = EnclaveManager(
+                    config, enclave_signup_data, extended_measurements)
+                # Persist updated worker to KV storage
+                persist_worker(enclave_manager, kv_helper)
+                key_refresh_time = 0
+
             # Poll KV storage for new work-order requests and process
             enclave_manager.process_work_orders(kv_helper)
             logger.info("Enclave manager sleeping for %d secs", sleep_interval)
             time.sleep(sleep_interval)
+            key_refresh_time = key_refresh_time + sleep_interval
```

Comment:
Decouple key refresh from work order execution. You could use asyncio coroutine to create key refresh as a separate task. That's will also avoid using sleep interval which is used for work order polling.

Reply:
I will explore asyncio and use it to do key refresh